torch/_meta_registrations.py

```diff
@@ -5784,6 +5784,26 @@ def _cudnn_rnn(
     return output, hy, cy, reserve, weight_buf
 
 
+@register_meta(aten._cudnn_rnn_flatten_weight.default)
+def _cudnn_rnn_flatten_weight(
+    weight_arr,
+    weight_stride0,
+    input_size,
+    mode,
+    hidden_size,
+    proj_size,
+    num_layers,
+    batch_first,
+    bidirectional,
+):
+    torch._check(len(weight_arr) > 0)
+    num_weights = sum(weight.numel() for weight in weight_arr)
+    weight_buf = weight_arr[0].new_empty(
+        num_weights, requires_grad=weight_arr[0].requires_grad
+    )
+    return weight_buf
```

Comment:
I believe this `_cudnn_rnn_flatten_weight` does in-place stride modifications to the input parameters. See, this comment here: https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/rnn.py#L203-L204

We would to in-place update the metas of the inputs.